code/lib/cli/src/automigrate/fixes/bare-mdx-stories-glob.test.ts

```diff
@@ -0,0 +1,205 @@
+      },
+      {
+        existingStoriesEntries: ['../src/**/*.stories.@(js|jsx|mdx|ts|tsx)'],
+        expectedStoriesEntries: ['../src/**/*.stories.@(js|jsx|ts|tsx)', '../src/**/*.mdx'],
+      },
+      {
+        existingStoriesEntries: ['../src/**/*.stories.@(js|jsx|ts|tsx|mdx)'],
+        expectedStoriesEntries: ['../src/**/*.stories.@(js|jsx|ts|tsx)', '../src/**/*.mdx'],
+      },
+    ])(
+      'strings: $existingStoriesEntries',
+      async ({ existingStoriesEntries, expectedStoriesEntries }) => {
+        const packageJson = {
+          dependencies: { '@storybook/react': '^7.0.0' },
+        };
+        const main = {
+          stories: existingStoriesEntries,
+        };
+        await expect(checkBareMdxStoriesGlob({ packageJson, main })).resolves.toMatchObject({
+          existingStoriesEntries,
+          nextStoriesEntries: expectedStoriesEntries,
+        });
+      }
+    );
+    it.each([
+      {
+        existingStoriesEntries: [{ directory: '../src/**', files: '*.stories.mdx' }],
+        expectedStoriesEntries: [{ directory: '../src/**', files: '*.mdx' }],
+      },
+      {
+        existingStoriesEntries: [{ directory: '../src/**', files: '*.stories.*' }],
+        expectedStoriesEntries: [
+          { directory: '../src/**', files: '*.stories.*' },
+          { directory: '../src/**', files: '*.mdx' },
+        ],
+      },
+      {
+        existingStoriesEntries: [
+          { directory: '../src/**', files: '*.stories.@(js|jsx|ts|tsx|mdx)' },
+        ],
+        expectedStoriesEntries: [
+          { directory: '../src/**', files: '*.stories.@(js|jsx|ts|tsx)' },
+          { directory: '../src/**', files: '*.mdx' },
+        ],
+      },
+    ])(
+      'specifiers: $existingStoriesEntries.0.files',
+      async ({ existingStoriesEntries, expectedStoriesEntries }) => {
+        const packageJson = {
+          dependencies: { '@storybook/react': '^7.0.0' },
+        };
+        const main = {
+          stories: existingStoriesEntries,
+        };
+        await expect(checkBareMdxStoriesGlob({ packageJson, main })).resolves.toMatchObject({
+          existingStoriesEntries,
+          nextStoriesEntries: expectedStoriesEntries,
+        });
+      }
+    );
+
+    it('prompts', () => {
+      const result = bareMdxStoriesGlob.prompt({
+        existingStoriesEntries: [
+          '../src/**/*.stories.@(js|jsx|mdx|ts|tsx)',
+          { directory: '../src/**', files: '*.stories.mdx' },
+        ],
+        nextStoriesEntries: [
+          '../src/**/*.mdx',
+          '../src/**/*.stories.@(js|jsx|ts|tsx)',
+          { directory: '../src/**', files: '*.mdx' },
+        ],
+      } as BareMdxStoriesGlobRunOptions);
+
+      expect(result).toMatchInlineSnapshot(`
+        "We've detected your project has one or more globs in your 'stories' config that matches .stories.mdx files:
+          [36m\\"../src/**/*.stories.@(js|jsx|mdx|ts|tsx)\\"[39m
```

Comment:
Maybe we need a helper for this kind of tests to strip out the control sequences?